### PR TITLE
feat: add new/pending account statuses (with history) and a config for initial status

### PR DESCRIPTION
**Pull request by samerbuna**

### CODE REVIEW

File: src/graphql/admin/root/mutation/account-update-status.ts [hunk 1]
```diff
@@ -14,23 +14,31 @@ const AccountUpdateStatusInput = GT.Input({
     status: {
       type: GT.NonNull(AccountStatus),
```
[the remark this answers is not in the document]
Comment (samerbuna) [reply src/graphql/admin/root/mutation/account-update-status.ts:15]: Do you mean removing the status field and relying on the history to get the current status? That'll require a db migration. I thought we get this change out first and do the migration in another PR.

[the remark this answers is not in the document]
Comment (nicolasburtey) [reply src/graphql/admin/root/mutation/account-update-status.ts:15]: yes that is the idea. not duplicating the status at the db layer.

File: src/app/users/login.ts [hunk 2]
```diff
@@ -120,7 +120,16 @@ export const loginWithKratos = async ({
...
 
   const network = BTC_NETWORK
-  return createToken({ uid: user.id, network, kratosUserId: kratosUserIdValid })
+
+  const account = await AccountsRepository().findByUserId(user.id)
+  if (account instanceof Error) return account
+
+  return createToken({
+    uid: user.id,
+    accountStatus: account.status.toUpperCase(),
```
Comment (nicolasburtey) [inline src/app/users/login.ts:129]: is `accountStatus` really necessary?

This is a mutable value in the backend, so including this value in the token will lead to stale token.

What is the logic for including this in the JWT, versus doing a call to the backend to know the up-to-date status?

File: src/graphql/admin/schema.graphql [hunk 3]
```diff
@@ -25,6 +25,8 @@ enum AccountLevel {
 enum AccountStatus {
   ACTIVE
   LOCKED
+  NEW
+  PENDING
```
Comment (nicolasburtey) [inline src/graphql/admin/schema.graphql:29]: is pending needed? is there a difference between `new` and `pending`?

in both case, if the account needs manual approval to get active, the action will be limited on the account between those 2 status, so maybe we don't need to separate them?

**Review by nicolasburtey — COMMENTED** [review]
some questions/comments

File: src/app/users/login.ts (hunk 2 above)
Comment (samerbuna) [reply src/app/users/login.ts:129]: My thinking here: a token represents the authority of an account. The status of that account is part of that.

For clients apps enforcing the account status, they'll need to check that status every time they acquire a token.

On tokens being stale, the problem there is that we allow them to live beyond their need.

File: src/graphql/admin/schema.graphql (hunk 3 above)
Comment (samerbuna) [reply src/graphql/admin/schema.graphql:29]: New: Account is fresh (no status verifications happened yet). The UI sends the user to the verification flow
Pending: A verification request has been made but the result is not back yet. The UI can either block the user or limit what they can do.

File: src/app/users/login.ts (hunk 2 above)
Comment (bodymindarts) [reply src/app/users/login.ts:129]: I also have doubts about this - lets discuss the trade offs at standup.

File: src/graphql/admin/root/mutation/account-update-status.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (samerbuna) [reply src/graphql/admin/root/mutation/account-update-status.ts:15]: Done

File: src/app/users/login.ts (hunk 2 above)
Comment (samerbuna) [reply src/app/users/login.ts:128]: I removed this and relied on the auth endpoint instead.

File: src/domain/accounts/index.types.d.ts [hunk 4]
```diff
@@ -39,6 +39,13 @@ type AccountContact = {
   transactionsCount: number
 }
 
+type AccountStatusHistory = Array<{
+  status: AccountStatus
+  updatedByUserId: UserId
+  updatedAt?: Date
```
[the remark this answers is not in the document]
Comment (samerbuna) [reply src/domain/accounts/index.types.d.ts:44]: Always the default value. No need to pass it when inserting a new status record

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**